nerfstudio/data/pixel_samplers.py

```diff
@@ -105,6 +105,7 @@ def collate_image_dataset_batch(self, batch: Dict, num_rays_per_batch: int, keep
         assert collated_batch["image"].shape[0] == num_rays_per_batch
 
         # Needed to correct the random indices to their actual camera idx locations.
+        indices = indices.to(batch["image_idx"].device)
```

Comment:
Is this necessary since the line 109 below basically does the same thing?

Reply:
yes, because after 109 the contents of indices[:, 0] does not equal to batch["image_idx"][c] if they are not on the same device.